### PR TITLE
[plugin-excel] Allow to validate Excel files from any source

### PR DESCRIPTION
**Pull request by ikalinin1**

### CODE REVIEW

File: docs/modules/plugins/pages/plugin-excel.adoc [hunk 1]
```diff
@@ -98,3 +99,57 @@ When I create temporary excel file with content:
 |value1|value2|
 and put path to scenario variable `path`
 ----
+
+=== Excel validations
+
+==== By sheet name
+Checks that excel has a sheet with name *name* and specified *records*
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-excel.adoc:106]:
```suggestion
Checks that excel has a sheet with name *name* and specified *records*.
```

File: docs/modules/plugins/pages/plugin-excel.adoc [hunk 2]
```diff
@@ -98,3 +99,57 @@ When I create temporary excel file with content:
...
+Then `${response-as-bytes}` contains excel sheet with name `Mapping` and records:
+{valueSeparator=!}
+|cellsRange|valueRegex             |
+!A4:B5     !(Product|Price)\d+\s*  !
+!B3        !Price                  !
+!C1:C5     !                       !
+----
+
+==== By sheet index
+Checks that excel has a sheet with index *index* and specified *records*
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-excel.adoc:132]:
```suggestion
Checks that excel has a sheet with index *index* and specified *records*.
```

File: vividus-plugin-excel/src/main/java/org/vividus/steps/ExcelResponseValidationSteps.java [hunk 3]
```diff
@@ -136,11 +140,11 @@ private void reportResults(Stream<String> ranges, List<Entry<CellValue, Optional
         });
     }
 
-    private IExcelSheetsExtractor getExtractor()
+    private IExcelSheetsExtractor getExtractor(DataWrapper excelDoc)
     {
         try
         {
-            return new ExcelSheetsExtractor(httpTestContext.getResponse().getResponseBody());
```
Comment (valfirst) [inline vividus-plugin-excel/src/main/java/org/vividus/steps/ExcelResponseValidationSteps.java:143]: - `httpTestContext` is not used anymore ->
- `ExcelResponseValidationSteps` -> `ExcelDocumentValidationSteps` ->
- `vividus-plugin-rest-api` dependency can be dropped, can't it?

File: docs/modules/plugins/pages/plugin-excel.adoc [hunk 4]
```diff
@@ -98,3 +99,57 @@ When I create temporary excel file with content:
...
 ----
+
+=== Excel validations
+
+==== By sheet name
+Checks that excel has a sheet with name *name* and specified *records*
+
+[source,gherkin]
+----
+Then `$excelDoc` contains excel sheet with name `$name` and records:$records
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-excel.adoc:110]: I recommend to do not put shortcuts anywhere: `doc` -> `document`

**Review by valfirst — CHANGES_REQUESTED** [review]

File: docs/modules/plugins/pages/plugin-excel.adoc [hunk 5]
```diff
@@ -98,3 +99,57 @@ When I create temporary excel file with content:
...
+!B3        !Price                  !
+!C1:C5     !                       !
+----
+
+==== By sheet index
+Checks that excel has a sheet with index *index* and specified *records*.
+
+[source,gherkin]
+----
+Then `$$excelDocument` contains excel sheet with index `$index` and records:$records
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-excel.adoc:136]:
```suggestion
Then `$excelDocument` contains excel sheet with index `$index` and records:$records
```

File: vividus-tests/src/main/resources/story/integration/ExcelSteps.story [hunk 6]
```diff
@@ -16,13 +17,30 @@ Then response contains excel sheet with index `0` and records:
...
 Scenario: Step verification 'Then response contains excel sheet with name `$name` and records:$records'
+!-- Deprecated
 Then response contains excel sheet with name `Mapping` and records:
 {valueSeparator=!}
 |cellsRange|valueRegex             |
 !A4:B5     !(Product|Price)\d+\s*  !
 !B3        !Price                  !
 !C1:C5     !                       !
 
+Scenario: Step verification 'Then `$$excelDocument` contains excel sheet with index `$index` and records:$records'
```
Comment (valfirst) [inline vividus-tests/src/main/resources/story/integration/ExcelSteps.story:28]:
```suggestion
Scenario: Step verification 'Then `$excelDocument` contains excel sheet with index `$index` and records:$records'
```

File: vividus-tests/src/main/resources/story/integration/ExcelSteps.story [hunk 7]
```diff
@@ -16,13 +17,30 @@ Then response contains excel sheet with index `0` and records:
...
+Then `${response-as-bytes}` contains excel sheet with index `0` and records:
+{valueSeparator=!}
+|cellsRange|valueRegex             |
+!A4:B5     !(Product|Price)\d+\s*  !
+!B3        !Price                  !
+!C1:C5     !                       !
+
+Scenario: Step verification 'Then `$$excelDocument` contains excel sheet with name `$name` and records:$records'
```
Comment (valfirst) [inline vividus-tests/src/main/resources/story/integration/ExcelSteps.story:36]:
```suggestion
Scenario: Step verification 'Then `$excelDocument` contains excel sheet with name `$name` and records:$records'
```

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — APPROVED** [review]

**Review by uarlouski — APPROVED** [review]

**Pull request merged**